src/main/java/org/plumelib/bcelutil/StackTypes.java

```diff
@@ -0,0 +1,111 @@
+package org.plumelib.bcelutil;
+
+import org.apache.bcel.generic.InstructionList;
+import org.apache.bcel.generic.MethodGen;
+import org.apache.bcel.generic.Type;
+import org.apache.bcel.verifier.structurals.Frame;
+import org.apache.bcel.verifier.structurals.LocalVariables;
+import org.apache.bcel.verifier.structurals.OperandStack;
+import org.apache.bcel.verifier.structurals.UninitializedObjectType;
+
+/*>>>
+import org.checkerframework.checker.lock.qual.*;
+import org.checkerframework.checker.nullness.qual.*;
+import org.checkerframework.dataflow.qual.*;
+*/
+
+/**
+ * Stores the types on the stack at each instruction (identified by byte code offset) in a method.
+ */
+public final class StackTypes {
+
+  boolean track_locals = true;
```

Comment:
Is variable `track_locals` needed?  You could just let `loc_arr` be null and test that directly.  I think it could make the code easier to understand.